input-explainer.md

```diff
@@ -280,6 +280,78 @@ When the canvas receives a `pointerdown` event an `XRInputSource` is created wit
 
 For each of these events the `XRInputSource`'s target ray must be updated to originate at the point that was interacted with on the canvas, projected onto the near clipping plane (defined by the `depthNear` attribute of the `XRSession`) and extending out into the scene along that projected vector.
 
+## Button and Axis State
+
+Some applications need more than point-and-click style interaction provided by the `select` events. For devices that make use of controllers with buttons and axes, more complete information about the state of those inputs can be observed via the `XRInputSource`'s `gamepad` attribute.
+
+If the `XRInputState` represents an XR controller device with buttons or axes the `gamepad` attribute is an instance of a [`Gamepad`](https://w3c.github.io/gamepad/#gamepad-interface) interface, and `null` otherwise. Examples of controllers that may expose their state this way include Oculus Touch, Vive wands, Oculus Go and Daydream controllers, or other similar devices. More traditional gamepads, such as XBox or Playstation controllers, should not be exposed using this interface, nor should tracked devices without discreet inputs, such as optical hand tracking.
+
+`Gamepad` instances reported in this way have several notable behavioral changes vs. the ones reported by `navigator.getGamepads()`:
+
+ - `Gamepad` instances connected to an `XRInputSource` must not be included in the array returned by `navigator.getGamepads()`.
+ - The `Gamepad`'s `id` attribute must be empty string. The `XRInputState`'s `renderId` attribute should be used instead to identify the device.
+ - The `Gamepad`'s `index` attribute must be `-1`.
+ - The `Gamepad`'s `connected` attribute must always be `true`.
+ - The `Gamepad`'s `mapping` attribute must be `xr-standard`.
+
+All other attributes behave as described in the [Gamepad](https://w3c.github.io/gamepad/) specification.
+
+The WebXR Device API introduces a new standard controller layout indicated by the `mapping` value of `xr-standard`. (Additional mapping variants may be added in the future if necessary.) This defines a specific layout for the inputs most commonly found on XR controller devices today. The following table describes the buttons/axes and their physical locations:
+
+| Button/Axis  | Location                |
+| ------------ | ----------------------- |
+| buttons[0]   | Primary trigger         |
+| buttons[1]   | Touchpad/Joystick click |
+| buttons[2]   | Grip/Secondary trigger  |
+| axes[0]      | Touchpad/Joystick X     |
+| axes[1]      | Touchpad/Joystick Y     |
```

Comment:
I very much like the idea of an `"xr-standard"` mapping with inactive slots for missing parts.  Even if we then pursue an app-level mapping mechanism as @NellWaliczek described, we'll be taking far less forward-compat and risk if that's built on top of a consistent web API mapping like this.

As we think through the details of this proposal vs. the #462 proposal, there are two separate design questions that we should tease apart:
* Do we make a new "tracked controller" API or reuse the `Gamepad` API? _(the vote at the F2F here was quite decisive)_
* Do we represent `trigger`/`touchpad`/`joystick`/`grip` separately, or semantically cluster them into "primary trigger", "touchpad/joystick" and "grip/secondary trigger" buckets? _(this was not yet addressed)_

In designing such an`"xr-standard"` mapping, an alternative here is to just go all the way and create a more detailed mapping that skips this semantic clustering.  For example, here, axes 0 and 1 are multiplexed across both touchpad and joystick, depending on what the device has.  Beyond the fact that we ship a device with both :smile:, I generally don't see app logic that expects either touchpad or joystick end up accidentally doing something reasonable when confronted with the opposite control.  For example, if apps are tested only on joysticks, they will often build interactions that expect the XY to start at (0, 0) for each interaction, which is not how touchpads work during use and often users end up blocked.

In the spirit of the original #462 proposal, we could instead establish explicit trigger, touchpad, joystick and primary controller button/axis IDs here, with UAs just leaving slots inactive when a control is not present on a device.  This creates a very predictable mapping and helps reduce the risk of per-UA bugs, since there are less judgment calls for each UA to make in deciding what is "primary" and "secondary".

One other risk of putting the semantic bucketing here is that it encourages app use of `buttons[0].pressed` rather than the `onselect` event.  This would be strictly worse for app compatibility, as it would exclude users of devices using hand-tracking or other non-controller XR input forms from proceeding.  If we want to establish an additional semantic bucket for secondary press, it may be better to just add `onsecondary` or such as a sibling to `onselect`, which also gives us an extra user-initiated event.

The counterbalancing risk with this more explicit approach is that an app developer who codes to this layer directly really does just think about one controller, mapping a key action to the "trigger" button ID even though the user's controller has no trigger.  However, if these low-level button/axis IDs did semantically cluster as proposed, but then a layer like @NellWaliczek described unpacks those values out anyway to strongly-named `trigger`, `touchpad` and `joystick` attributes, developers using that higher-level input API would fall into the same situation anyway.  It may be cleaner as everyone experiments with higher-level input abstractions to just make the low-level `"xr-standard"` mapping here very explicit about well-known parts.